flask/helpers.py

```diff
@@ -553,7 +538,9 @@ def send_file(filename_or_fp, mimetype=None, as_attachment=False,
         rv.cache_control.max_age = cache_timeout
         rv.expires = int(time() + cache_timeout)
 
-    if add_etags and filename is not None:
+    if add_etags and filename is not None and file is None:
```

Comment:
I removed it in master: https://github.com/pallets/flask/commit/01081dbe6cdfa3fc43d8e1fff708d4ed95e1be7e